### PR TITLE
Add ObjectEncoding command to JedisCluster

### LINKED ISSUE #2099: JedisClusterCommands doesn't implement objectEncoding ?
There is no support for objectEncoding in JedisCluster  ... can u please add one

### PR DESCRIPTION
**Pull request by MichalCho**
Resolves #2099

### CODE REVIEW

File: src/test/java/redis/clients/jedis/tests/commands/ClusterAdvancedJedisCommandsTest.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+  }
+
+  @Test
+  public void objectEncoding() {
+
+    jedisCluster.set("myKey","1");
+    assertEquals("int",jedisCluster.objectEncoding("myKey"));
+    jedisCluster.del("myKey");
+
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/ClusterAdvancedJedisCommandsTest.java:109]:
```suggestion
     assertNull(jedisCluster.objectEncoding("myKey")); 
```

File: src/test/java/redis/clients/jedis/tests/commands/ClusterAdvancedJedisCommandsTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+public class ClusterAdvancedJedisCommandsTest {
+
+  private Jedis node1;
+  private static Jedis node2;
+  private static Jedis node3;
+
+  private HostAndPort nodeInfo1 = HostAndPortUtil.getClusterServers().get(0);
+  private HostAndPort nodeInfo2 = HostAndPortUtil.getClusterServers().get(1);
+  private HostAndPort nodeInfo3 = HostAndPortUtil.getClusterServers().get(2);
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/ClusterAdvancedJedisCommandsTest.java:27]:
```suggestion
  private final HostAndPort nodeInfo1 = HostAndPortUtil.getClusterServers().get(0);
  private final HostAndPort nodeInfo2 = HostAndPortUtil.getClusterServers().get(1);
  private final HostAndPort nodeInfo3 = HostAndPortUtil.getClusterServers().get(2);
```

File: src/test/java/redis/clients/jedis/tests/commands/ClusterAdvancedJedisCommandsTest.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  private final Set<HostAndPort> jedisClusterNode = new HashSet<>();
+  JedisCluster jedisCluster;
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/ClusterAdvancedJedisCommandsTest.java:29]:
```suggestion
  private JedisCluster jedisCluster;
```

File: src/main/java/redis/clients/jedis/commands/AdvancedClusterCommands.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package redis.clients.jedis.commands;
+
+import redis.clients.jedis.params.ClientKillParams;
+import redis.clients.jedis.params.MigrateParams;
+import redis.clients.jedis.util.Slowlog;
+
+import java.util.List;
```
Comment (gkorland) [inline src/main/java/redis/clients/jedis/commands/AdvancedClusterCommands.java:7]:
```suggestion
```

File: src/test/java/redis/clients/jedis/tests/commands/ClusterAdvancedJedisCommandsTest.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisCluster;
+import redis.clients.jedis.JedisPoolConfig;
+import redis.clients.jedis.tests.HostAndPortUtil;
+import redis.clients.jedis.util.JedisClusterCRC16;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/ClusterAdvancedJedisCommandsTest.java:17]:
```suggestion
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;
```

File: src/test/java/redis/clients/jedis/tests/commands/ClusterAdvancedJedisCommandsTest.java [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+public class ClusterAdvancedJedisCommandsTest {
+
+  private Jedis node1;
+  private static Jedis node2;
+  private static Jedis node3;
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/ClusterAdvancedJedisCommandsTest.java:23]:
```suggestion
  private Jedis node2;
  private Jedis node3;
```

**Review by gkorland — CHANGES_REQUESTED** [review]

**Comment by sazzad16** [comment]
Closed by https://github.com/redis/jedis/commit/858c8051301854e8343eef66aa68c6b9dab1c328

**Pull request closed without merging**